### PR TITLE
bugfix: check pc.c.XDSUpdater instead of pc.c

### PR DESCRIPTION
**Pull request by YaoZengzeng**
Signed-off-by: YaoZengzeng <yaozengzeng@zju.edu.cn>

### CODE REVIEW

File: pilot/pkg/serviceregistry/kube/pod.go [hunk 1]
```diff
@@ -107,7 +107,7 @@ func (pc *PodCache) event(obj interface{}, ev model.Event) error {
 			// delete only if this pod was in the cache
 			if pc.keys[ip] == key {
 				delete(pc.keys, ip)
-				if pc.c != nil {
+				if pc.c.XDSUpdater != nil {
```
Comment (ymesika) [inline pilot/pkg/serviceregistry/kube/pod.go]: Can you double check this? `pc.c` is a reference field and can be nil.
Maybe all the if conditions above (`if pc.c.XDSUpdater != nil`) are the incorrect ones.

**Comment by YaoZengzeng** [comment]
@ymesika Updated

Would mind reply with the `ok-to-test` to trigger the test? :)

**Comment by YaoZengzeng** [comment]
@ymesika Please re-trigger the failed flake CI case to get it passed 

And reply with `/lgtm` and `/approve`

Otherwise the PR will not be automatically merged. Thanks! :)

**Comment by hzxuzhonghu** [comment]
Is there any case pc.c == nil ?

**Comment by YaoZengzeng** [comment]
@hzxuzhonghu Now seems not, but double check will always be right.

**Comment by ymesika** [comment]
Inspecting the call hierarchy it seems like `pc.c` will never be nil but I'm fine with verifying it.

/lgtm
/approve

**Pull request merged**